### PR TITLE
fix(lib)!: align go compatibility edge cases

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- fix exact large-integer ordering in `max` and `min`
- preserve integer precision in the differential-test comparator
- support negative `get` indices and numeric float sorting
- align `repeat`, counted `replace`, pipe argument order, and `reduce` forms with Go expr
- make `flatten` iterative so deeply nested arrays cannot exhaust the native stack
- add shared positive and error corpora verified by both implementations

## Why

The v2 compatibility stack exposed edge cases that were not represented in the original positive-only corpus. This PR addresses all actionable unresolved correctness feedback on #87, #88, and #89 and adds regression coverage for older builtin mismatches.

## Impact

Existing incorrect compatibility behavior changes to match Go expr v1.17.8. Mixed unorderable values now return an expression error instead of being sorted by their display strings. Empty `reduce` calls without an initial value now error instead of returning `nil`.

## Stack

- built on #89
- merge the preceding compatibility and builtin PRs first

## Checks

- Go expr v1.17.8 verified 102 result lines and 5 error cases
- `cargo test --all-features` (115 unit tests, 2 compatibility tests, 10 doctests)
- 20,000-level iterative `flatten` regression test
- `git diff --check`

_This PR description was generated by OpenAI Codex._

---

> [!NOTE]
> **Medium Risk**
> Changes expression semantics and parsing (breaking for mismatched `repeat`, `sort`, and `reduce`), but scope is bounded to Go-compat alignment with regression corpora on both sides.
> 
> **Overview**
> **Aligns runtime behavior and parsing with Go expr v1.17.8**, with shared positive (`cases.tsv`) and negative (`errors.tsv`) corpora checked by both the Go verifier and Rust `go_compat` tests.
> 
> **Builtin semantics:** `repeat` now repeats exactly *n* times (fixes prior off-by-one), rejects negative counts, and enforces a memory cap; `replace` accepts an optional fourth argument for limited replacements. **`sort` / `sortBy`** require homogeneous numeric or string keys—mixed types and NaN error instead of falling back to string comparison—and compare ints and floats numerically. **`flatten`** is rewritten iteratively to avoid stack overflow on deep nesting. **`get`** on arrays supports negative indices (matching index syntax).
> 
> **Grammar:** function-call parsing allows ordinary `expr` arguments before an optional predicate and additional trailing args, enabling forms like `reduce([...], {#acc + #}, 10)` and pipe `reduce` with braced reducers.
> 
> **Breaking:** expressions that previously “worked” with wrong semantics (e.g. empty `reduce` without an initial value, invalid `sort` mixes) now fail like Go expr.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e23dee8267c26f07474221b260384c416a2abb7b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/functions/array.rs [hunk 1]
```diff
@@ -326,18 +326,21 @@ pub fn add_array_functions(env: &mut Environment) {
...
+            });
             if desc {
                 cmp.reverse()
             } else {
                 cmp
             }
         });
+        if let Some(error) = sort_error {
+            bail!("sort() {error}");
+        }
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/functions/array.rs]: Fixed in 6d0f443 by validating all sort values and sortBy keys before sorting, then using an infallible comparator. Added a three-element mixed-type error-corpus regression verified against Go expr. _This comment was generated by OpenAI Codex._
